src/batch/src/executor/project_set.rs

```diff
@@ -95,14 +95,18 @@ impl ProjectSetExecutor {
                 let items = results
                     .iter()
                     .map(|result| match result {
-                        Either::Left(arrays) => Either::Left(arrays[row_idx].clone()),
+                        Either::Left(array) => Either::Left(array.value_at(row_idx)),
                         Either::Right(array) => Either::Right(array.value_at(row_idx)),
                     })
                     .collect_vec();
                 // The maximum length of the results of table functions will be the output length.
                 let max_tf_len = items
                     .iter()
-                    .map(|i| i.as_ref().map_left(|arr| arr.len()).left_or(0))
+                    .map(|i| {
+                        i.as_ref()
+                            .map_left(|arr| arr.map_or(0, |arr| arr.len()))
```

Comment:
When will we get `NULL` list? 👀 Are the semantics the same as an empty list?

Reply:
A NULL input will generate a NULL list. A non-NULL input may generate an empty list, e.g. `generate_series(1, 0, 1)`.
I guess they are the same in subsequent processing.